### ISSUE TITLE
Allow to remove an item from within the `renderValue` function of MultiSelect

### ISSUE THREAD

**Issue by jcmonnin** [kobaltedev/kobalte#148]
The use case for this is to implement a MultiSelect where the values in the trigger are rendered as bage with a close button to remove/unselect the item. This is how the Hope-UI multi-select was styled.

The click handler of the close button must be able to unselect the item. This currently only possible in a controlled select where unselect is handled outside of the Kobalte component.

Following implementation would allow to unselect an item directly from the `renderValue` function:

- The `selectedOptions` adds an additional `onRemove` property than can be used in the button handler
- A 2nd argument is supplied to the `renderValue` function. The signature would become something like:
  `renderValue: (selectedOptions: Accessor<any[]>, onRemove: (value: string) => void) => JSX.Element;`

There is a discussion in the kobalte group on discord about it. Note that it's possible to work around the issue by using a controlled select. When building a styled component that needs to be usable in both controlled and uncontrolled mode, the `createControllableSetSignal` can be used as in:
https://github.com/kobaltedev/kobalte/blob/main/packages/core/src/select/select-root.tsx#L41

**Comment by jcmonnin** [comment]
There is not much difference from your example to think as the value renderer as it's own component. Something like this:

```
const MyMultiSelectValue = (props) => {
  return (
    <>
      <For each={props.items}>
        {item => (
          <span>
            {item.rawValue} // item is now of type CollectionNode<T> hense the use of `rawValue`
            <button
              // prevent closing the select.
              onPointerDown={e => e.stopPropagation()}
              onClick={() => props.remove(item)} // remove one item
            >
              X
            </button>
          </span>
        )}
      </For>
      <button
        // prevent closing the select.
        onPointerDown={e => e.stopPropagation()}
        onClick={props.clear} // remove all items
      >
        X
      </button>
    </>
  );
}


<MultiSelect.Root
  options={["Apple", "Banana", "Blueberry", "Grapes", "Pineapple"]}
  placeholder="Select some fruits…"
  renderValue={MyMultiSelectValue}
>
  ...
</MultiSelect.Root>
```
Note that `selection` becomes `props`, and `props.items` is a property, not a function. This is to follow the convention for components, where props are passed as values with getter function. I'm not saying this is necessary, but possibly having the option to write a normal component is more readable.

**Comment by fabien-ml** [comment]
If you go this way I suggest to do it like [mantine](https://mantine.dev/core/multi-select/#custom-label-component) and use the same API for item and section too :

```tsx
function Value(props) {
  return (
    <>
      <For each={props.items}>
        {item => (
          <span>
            {item.rawValue} // item is now of type CollectionNode<T> hense the use of `rawValue`
            <button
              // prevent closing the select.
              onPointerDown={e => e.stopPropagation()}
              onClick={() => props.remove(item)} // remove one item
            >
              X
            </button>
          </span>
          )}
      </For>
      <button
        // prevent closing the select.
        onPointerDown={e => e.stopPropagation()}
        onClick={props.clear} // remove all items
      >
        X
      </button>
    </>
  );
}

function Item(props) {
  return (
    <Select.Item {...props}>
      <Select.ItemLabel>{props.item.rawValue}</Select.ItemLabel>
      <Select.ItemIndicator>
        <CheckIcon />
      </Select.ItemIndicator>
    </Select.Item>
  );
}

function Section(props) {
  return <Select.Section {...props}>{props.section.rawValue.label}</Select.Section>;
}

<MultiSelect.Root
  options={["Apple", "Banana", "Blueberry", "Grapes", "Pineapple"]}
  placeholder="Select some fruits…"
  valueComponent={Value}
  itemComponent={Item}
  sectionComponent={Section}
>
  <MultiSelect.Trigger>
    <MultiSelect.Value />
    <MultiSelect.Icon>
      <CaretSortIcon />
    </MultiSelect.Icon>
  </MultiSelect.Trigger>
  <MultiSelect.Portal>
    <MultiSelect.Content>
      <MultiSelect.Listbox />
    </MultiSelect.Content>
  </MultiSelect.Portal>
</MultiSelect.Root>
```

The confusing part is that `Value` get rendered inside `MultiSelect.Value`.

Another option may be passing a render prop directly to `MultiSelect.Value`, but it's like my 1st option, we've just moved the problem :
```tsx
function Item(props) {
  return (
    <Select.Item {...props}>
      <Select.ItemLabel>{props.item.rawValue}</Select.ItemLabel>
      <Select.ItemIndicator>
        <CheckIcon />
      </Select.ItemIndicator>
    </Select.Item>
  );
}

function Section(props) {
  return <Select.Section {...props}>{props.section.rawValue.label}</Select.Section>;
}

<MultiSelect.Root
  options={["Apple", "Banana", "Blueberry", "Grapes", "Pineapple"]}
  placeholder="Select some fruits…"
  valueComponent={Value}
  itemComponent={Item}
  sectionComponent={Section}
>
  <MultiSelect.Trigger>
    <MultiSelect.Value>
      {selection => ({/* render your thing */})}
    </MultiSelect.Value>
    <MultiSelect.Icon>
      <CaretSortIcon />
    </MultiSelect.Icon>
  </MultiSelect.Trigger>
  <MultiSelect.Portal>
    <MultiSelect.Content>
      <MultiSelect.Listbox />
    </MultiSelect.Content>
  </MultiSelect.Portal>
</MultiSelect.Root>
```

**Comment by fabien-ml** [comment]
fixed in `v0.8.0`.
